components/context-menu/utils/keyboard-input-manager.js

```diff
@@ -149,6 +149,10 @@ export class KeyboardInputManager {
     async #setTabIndex(element, siblingType = null) {
```

Comment:
this function is way to complex for what its doing. please take step back and improve